### DISCUSSION TITLE
Disabling attributes from instrumentation

### DISCUSSION THREAD

**Discussion by ecourreges-orange** [open-telemetry/opentelemetry-java-instrumentation#2204] [Q&A] [answered]
Hi,
I know I can fully disable instrumentation of classes or modules, but what if I want to **disable attributes** for saving storage/network/performance/security reasons?

For example, I would like to have jdbc enabled, but filter out the `db.statement` to avoid exposing too much but still knowing that statements to a certain table are made through the span name.

Is the only way to do this through an attribute filter in the collector? 
Or can it be done at the source of the information?

Thank you.

**Comment by iNikem** [comment, marked as answer]
Currently no, instrumentations do not provide such functionality.

**Comment by anuraaga** [comment]
@ecourreges-orange Just curious, are you able to use the opentelemetry collector? It has a feature for filtering attributes which may work. It's currently our way for filtering attributes.

**Reply by ecourreges-orange** [reply]
Yes indeed, I am using the opentelemetry-collector but in a central location, so will probably setup filtering for `db.statement` and maybe other attributes for everyone.
